src/z2m/mod.rs

```diff
@@ -188,6 +188,29 @@ impl Client {
             room_name = &grp.friendly_name;
         }
 
+        let topic = grp.friendly_name.to_string();
+        let mut metadata = RoomMetadata::new(RoomArchetype::Home, room_name, false);
+        if let Some(room_conf) = self.config.rooms.get(&topic) {
+            if let Some(name) = &room_conf.name {
+                metadata.name = name.to_string();
+            }
+            if let Some(icon) = &room_conf.icon {
+                metadata.archetype = *icon;
+            }
+            if let Some(hidden) = &room_conf.hidden {
+                metadata.hidden = *hidden;
+            }
+        };
+
+        if metadata.hidden {
+            log::debug!(
+                "[{}] ({}) is hidden; passing on link!",
+                self.name,
+                room_name
+            );
+            return Ok(());
```

Comment:
good to have debug feedback. Can we say something other than "passing on link"? If I hadn't written the rest of the surrounding code, I don't think I would know what that meant :)